### PR TITLE
Edit offer

### PR DESCRIPTION
**Pull request by Rantaa**

### CODE REVIEW

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java [hunk 1]
```diff
@@ -142,7 +162,7 @@ public void setCurrentdeed(UUID id) {
     /**
      * @return the currently assigned currentDeed.
      */
-    public IDeed getCurrentDeed() {
+    public Deed getCurrentDeed() {
```
Comment (antjanne) [inline goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java]: Why Deed, might be a dumb question. 👍

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/View/ViewDeed.java [hunk 2]
```diff
@@ -20,6 +23,17 @@ public void onCreate(Bundle savedInstanceState) {
         subject.setText(deed.getSubject());
         TextView description = findViewById(R.id.deedDescription);
         description.setText(deed.getDescription());
+        View button = (Button) findViewById(R.id.edit_offer);
+        if (!deedController.isMyActiveDeedHandler()) {
+            button.setVisibility(View.GONE); }
 
     }
+
```
Comment (richardmarklund) [inline goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/View/ViewDeed.java]: you maybe should use code formatter and maybe the on save-plugin we talked about at the start of the course. You have lots of spaces and stuff thats not necessary

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java [hunk 3]
```diff
@@ -15,20 +15,24 @@
  * and the current logged in account.
  * */
 
+
 public class GoodDeeds {
 
 
     private static GoodDeeds goodDeeds;
```
[the remark this answers is not in the document]
Comment (Rantaa) [reply goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java]: Yep, I would argue that it's alright to use the entity inside the model.. also, I needed to handle an actual deed for setting the subject/description.

File: goodDeeds/app/src/test/java/com/goodpeople/gooddeeds/Model/GoodDeedsTest.java [hunk 4]
```diff
@@ -1,149 +1,185 @@
...
+    public  void testEditOnlyYourOwnDeeds() {
+        String subject = deed.getSubject();
+        String description = deed.getDescription();
+        gd.createAccount("SomeoneElse", 00000, "test@test.com", "123");
+        IAccount newAccount = gd.getAccounts().get(1);
+        Deed d = (Deed) deed;
+        d.setGivingAccount(newAccount);
+        gd.editOffer("Changed", "Changed");
+        assertSame(subject, deed.getSubject());
+        assertSame(description, deed.getDescription());
```
Comment (richardmarklund) [inline goodDeeds/app/src/test/java/com/goodpeople/gooddeeds/Model/GoodDeedsTest.java]: Change this to :
    @Test
    public  void testEditOnlyYourOwnDeeds() {
        gd.createAccount("SomeoneElse", 00000, "test@test.com", "123");
        IAccount newAccount = gd.getAccounts().get(1);
        Deed d = (Deed) deed;
        d.setGivingAccount(newAccount);
        gd.editOffer("Changed", "Changed");
        assertSame("Changed", deed.getSubject());
        assertSame("Changed", deed.getDescription());

    }

And all tests go through

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java [hunk 5]
```diff
@@ -13,25 +13,29 @@
...
-        Account a = new Account("Anton",30597,"anton46304@gmail.com","ahah");
-        Deed d = Deed.newOffer(a,"Gräsklipp","Jag hjälper gärna till att klippa gräsmattan i storgöteborg, ge mig en pling");
-        Deed d2 = Deed.newOffer(a,"Hårklipp","Jag klipper gärna håret på folk! Ge mig en pling vetja!");
+
+
+        Account a2 = new Account("Anton", 30597, "1234@gmail.com", "ahah");
+
+        Account a = new Account("Anton", 30597, "anton46304@gmail.com", "ahah");
+        Deed d = Deed.newOffer(a, "Gräsklipp", "Jag hjälper gärna till att klippa gräsmattan i storgöteborg, ge mig en pling");
+        Deed d2 = Deed.newOffer(a, "Hårklipp", "Jag klipper gärna håret på folk! Ge mig en pling vetja!");
```
Comment (richardmarklund) [inline goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/Model/GoodDeeds.java]: Comment out mock data

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/View/EditOffer.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        editSubj.setText(deedController.getDeedSubject());
+        editDesc.setText(deedController.getDeedDescription());
+        submitButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                EditText subjectEditText = findViewById(R.id.subjectEditText);
+                EditText descriptionEditText = findViewById(R.id.descriptionEditText);
+                deedController.editOfferHandler(subjectEditText.getText().toString(), descriptionEditText.getText().toString());
+                //View desc = findViewById(R.id.view_offer);
+                //desc.requestLayout();
```
Comment (richardmarklund) [inline goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/View/EditOffer.java]: remove commented out code

File: goodDeeds/app/src/test/java/com/goodpeople/gooddeeds/Model/GoodDeedsTest.java (hunk 4 above)
Comment (Rantaa) [reply goodDeeds/app/src/test/java/com/goodpeople/gooddeeds/Model/GoodDeedsTest.java]: Solved

File: goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/View/EditOffer.java (hunk 6 above)
Comment (Rantaa) [reply goodDeeds/app/src/main/java/com/goodpeople/gooddeeds/View/EditOffer.java]: Solved

**Pull request merged**